ambry-api/src/main/java/com.github.ambry/clustermap/PartitionStateChangeListener.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2019 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.clustermap;
+
+/**
+ * {@link PartitionStateChangeListener} takes action when partition state changes.
+ */
+public interface PartitionStateChangeListener {
+
+  /**
+   * Action to take on becoming leader of a partition.
+   * @param partitionName of the partition.
+   */
+  void onPartitionLeadFromStandby(String partitionName);
+
+  /**
+   * Action to take on being removed as leader of a partition.
+   * @param partitionName of the partition.
+   */
+  void onPartitionStandbyFromLead(String partitionName);
```

Comment:
Same here, better to rename it to `onPartitionBecomeStandbyFromLeader`.
@lightningrob there might be some ambiguity between participant and partition. For partition, yes, it may have other transitions between states we defined. I will add rest of methods in this class because `PartitionStateChangeListener` will be used for "Move Replica" as well. We need to add/remove replicas on current node when state transition occurs from OFFLINE to BOOTSTRAP or from STANDBY to INACTIVE. 

Participant is basically datanode, it has multiple partitions and they may stay in different states. The state model applies to partition only.